src/octoprint/static/js/app/viewmodels/files.js

```diff
@@ -423,16 +424,74 @@ $(function() {
                     }
                 }
                 output += gettext("Estimated Print Time") + ": " + formatDuration(data["gcodeAnalysis"]["estimatedPrintTime"]) + "<br>";
-		if(self.loginState.isUser()&&data["gcodeAnalysis"]["warning"])
+		if(self.loginState.isUser())
 		{
-		    var warning = "<p>" + gettext("Object exceeds printing area") + "</p>";
-                    warning += pnotifyAdditionalInfo("<pre>Object or objects in file '"+data["name"]+"' exceed/s the printing area</pre>");
-                    new PNotify({
-                        title: "Object dimensions",
-                        text: warning,
-                        type: "warning",
-                        hide: false
-                    });
+                    var profileItems=self.printerProfiles.profiles.items();
+                    var count=0;
+                    while(count<profileItems.length && !profileItems[count].current)
+                    {
+                        count++;
+                    }
+                    if(count<profileItems.length)
+                    {
+                        var warn=false;
+                        var volumeInfo=profileItems[count].volume;
+                        var printingArea=data["gcodeAnalysis"]["printingArea"];
+                        if(volumeInfo.origin=="lowerleft")
+                        {
+                            if(printingArea["maxX"]>volumeInfo.width || printingArea["maxY"]>volumeInfo.depth || printingArea["maxZ"]>volumeInfo.height)
+                            {
+                                warn=true;
+                            }
+                        }
+                        else //origin is centre
+                        {
+                            if( printingArea["maxX"]>(volumeInfo.width/2) || printingArea["maxY"]>(volumeInfo.depth/2) || printingArea["maxZ"]>volumeInfo.height || Math.abs(printingArea["minX"])>(volumeInfo.width/2) || Math.abs(printingArea["minY"])>(volumeInfo.depth/2))
+                            {
+                                warn=true;
+                            }
+                        }
+                    }
+                    if(warn)
```

Comment:
Ufff... Yes, that does what it's supposed to do so, but it is way too repetitive and as such highly error prone in case of future adaption/bug fixing.

Take a closer look at your `if` conditions. Most of them start with either testing for origin type `lowerleft` vs. `center`.

So we can extract those tests already (also, please take more care of the code formatting - space between if and condition, also within expressions, starting brackets on the same line, *consistent* indentation):

``` javascript
if (volumeInfo.origin == "lowerleft") {
    if (printingArea["maxX"] > volumeInfo.width) {
        // ...
    }
    if (printingArea["maxY"] > volumeInfo.depth) {
        // ...
    }
} else if (volumeInfo.origin == "center") {
    if (printingArea["maxX"] > (volumeInfo.width / 2)) {
        // ...
    }
    if (printingArea["minX"] < (-volumeInfo.width / 2)) {
        // ...
    }
    if (printingArea["maxY"] > (volumeInfo.depth / 2)) {
        // ...
    }
    if (printingArea["minY"] < (-volumeInfo.depth / 2)) {
        // ...
    }
}

if (printingArea["maxZ"] > volumeInfo.height) {
    // ...
}
```

Now, take a look at the two origin-controlled blocks. Still look quite similar. By extracting the min/max values of the volume into custom vars we can make the check even easier:

``` javascript
// set print volume boundaries
var boundaries = {
    minX = 0,
    maxX = volumeInfo.width,
    minY = 0,
    maxY = volumeInfo.depth,
    minZ = 0,
    maxZ = volumeInfo.height
};
if (volumeInfo.origin == "center") {
    boundaries["maxX"] = volumeInfo.width / 2;
    boundaries["minX"] = -1 * boundaries["maxX"];
    boundaries["maxY"] = volumeInfo.depth / 2;
    boundaries["minY"] = -1 * boundaries["maxY"];
}

// find exceeded dimensions
if (printingArea["minX"] < boundaries["minX"] || printingArea["maxX"] > boundaries["maxX"]) {
    info += "Profile's width: (" + boundaries["minX"] + ", " + boundaries["maxX"] + ") vs object's width: (" + printingArea["minX"] + ", " + printingArea["maxX"] + ")\n";
}
if (printingArea["minY"] < boundaries["minY"] || printingArea["maxY"] > boundaries["maxY"]) {
    info += "Profile's depth: (" + boundaries["minY"] + ", " + boundaries["maxY"] + ") vs object's depth: (" + printingArea["minY"] + ", " + printingArea["maxY"] + ")\n";
}
if (printingArea["minZ"] < boundaries["minZ"] || printingArea["maxZ"] > boundaries["maxZ"]) {
    info += "Profile's height: (" + boundaries["minZ"] + ", " + boundaries["maxZ"] + ") vs object's width: (" + printingArea["minZ"] + ", " + printingArea["maxZ"] + ")\n";
}
```